docs/topics/numbers.md

```diff
@@ -0,0 +1,295 @@
+[//]: # (title: Numbers)
+
+## Integer types
+
+Kotlin provides a set of built-in types that represent numbers.  
+For integer numbers, there are four types with different sizes and, hence, value ranges.
+
+| Type	 |Size (bits)| Min value| Max value|
+|--------|-----------|----------|--------- |
+| `Byte`	 | 8         |-128      |127       |
+| `Short`	 | 16        |-32768    |32767     |
+| `Int`	 | 32        |-2,147,483,648 (-2<sup>31</sup>)| 2,147,483,647 (2<sup>31</sup> - 1)|
+| `Long`	 | 64        |-9,223,372,036,854,775,808 (-2<sup>63</sup>)|9,223,372,036,854,775,807 (2<sup>63</sup> - 1)|
+
+If you initialize a variable with an integer value not exceeding the maximum value of `Int`, the compiler will 
+automatically infer the type `Int`. If the initial value exceeds the maximum value, then the type is `Long`.
+To specify the `Long` value explicitly, append the suffix `L` to the value.
+
+```kotlin
+val one = 1 // Int
+val threeBillion = 3000000000 // Long
+val oneLong = 1L // Long
+val oneByte: Byte = 1
+```
+
+> Inside the compiler, there is a priority between built-in integer types. If there is no explicit type specification
+> (such as `val oneByte: Byte = 1`), an integer literal that is in the value range of multiple built-in integer types 
+> receives the type of the highest priority. Thus, the integer literal `1` in the first line of the example receives 
+> the `Int` type, although it can be either `Byte` or `Short`.
```

Comment:
I would say it more shortly without mentioning internal priority in the compiler:
> If there is no explicit type specification and no explicit `L` suffix, the type is inferred automatically as `Int` or `Long` based on the value of an integer literal: the type with the smallest range enough to represent the value is chosen.

Reply:
At this point it merely restates the text before the example in other words. Probably Roman proposed to say something about the situation when the expected type is specified explicitly? I.e. when the expected type is Byte, the literal becomes the literal of type Byte and is checked not to exceed the range of Byte type.

Reply:
Thanks, I removed the note and tried to compose all this information in the paragraph above.
Please see [the commit](https://github.com/JetBrains/kotlin-web-site/pull/2409/commits/8b4528422f9620a7fcc76c179dc780166ed29bbd).